cas-server-core/src/main/java/org/jasig/cas/logout/LogoutManagerImpl.java

```diff
@@ -118,15 +118,7 @@ public void setIssueAsynchronousCallbacks(final boolean asyncCallbacks) {
      */
     @Override
     public List<LogoutRequest> performLogout(final TicketGrantingTicket ticket) {
-        final Map<String, Service> services;
-        // synchronize the retrieval of the services and their cleaning for the TGT
-        // to avoid concurrent logout mess ups
-        synchronized (ticket) {
-            services = ticket.getServices();
-            ticket.removeAllServices();
-        }
-        ticket.markTicketExpired();
-
+        final Map<String, Service> services = ticket.getServices();
```

Comment:
The similar locking issue is here:
https://github.com/Jasig/cas/blob/4.1.x/cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java#L437
But it seems to be separate topic.